src/bus/mod.rs

```diff
@@ -46,30 +46,40 @@ enum Listener {
 
 impl Bus {
     pub async fn for_address(address: Option<&str>) -> Result<Self> {
-        let mut address = match address {
+        let address = match address {
             Some(address) => Address::from_str(address)?,
             None => Address::from_str(&default_address())?,
         };
-        let guid: OwnedGuid = match address.guid() {
-            Some(guid) => guid.to_owned().into(),
-            None => {
-                let guid = Guid::generate();
-                address = address.set_guid(guid.clone())?;
-
-                guid.into()
-            }
+        let address = if address.guid().is_some() {
+            address
+        } else {
+            let guid = Guid::generate();
+            address.set_guid(guid.clone())?
+        };
+        let guid: OwnedGuid = address
+            .guid()
+            .expect("Address key (guid) should already be initialized.")
+            .to_owned()
+            .into();
+
+        #[cfg(unix)]
+        let address = if let Transport::Unix(unix) = address.transport() {
```

Comment:
I'm not sure this is a good change. We end up matching the same exact thing twice one after another. An extra level of indentation is just a price you've to pay for `match`.